src/store/userSlice.js

```diff
@@ -15,31 +15,52 @@ const userSlice = createSlice({
 	},
 });
 export default userSlice.reducer;
-export const { login } = userSlice.actions;
+export const { login, logout } = userSlice.actions;
+
 export const attemptTokenLogin = () => async (dispatch) => {
 	const token = window.localStorage.getItem("token");
 	if (token) {
-		const { data: userInfo } = await axios.get("/api/auth", {
-			headers: {
-				authorization: token,
-			},
-		});
+		const { data: userInfo } = await axios.post(
+			"/api/auth",
+			{},
+			{
+				headers: {
+					authorization: token,
+				},
+			}
+		);
 		dispatch(login(userInfo));
 	}
-	return false;
 };
-export const attemptPssWordLogin = (loginInfo) => async (dispatch) => {
+export const attemptPsswordLogin = (loginInfo) => async (dispatch) => {
 	try {
-		const { data: token } = await axios.post("/api/user/login", loginInfo);
+		console.log("slice", loginInfo);
+		const { data: token } = await axios.post("/api/auth/login", loginInfo);
+		console.log(token);
 		window.localStorage.setItem("token", token);
-		attemptTokenLogin()(dispatch);
+		dispatch(attemptTokenLogin());
 	} catch (error) {
 		return error;
 	}
 };
 export const logoutUser = () => {
 	return (dispatch) => {
 		dispatch(logout());
-		window.localStorage.setItem("token", "");
+		window.localStorage.removeItem("token");
+	};
+};
+export const createUser = (userDetails, isAdmin = false) => {
+	return async (dispatch) => {
+		try {
+			const { data: user } = await axios.post("/api/users", {
+				...userDetails,
+				isAdmin,
+			});
+			if (user) {
+				attemptPsswordLogin(user)(dispatch);
```

Comment:
need to change what we're passing to  attemptPsswordLogin; the user model returned has the password already passed.

I tried out attemptPsswordLogin({username: userDetails.username, password: userDetails.password})(dispatch), which then worked as expected